site/content/blog/release-0.30.0.md

```diff
@@ -0,0 +1,174 @@
+the cover of "this is a pre-1.0 release" and with the aim of pushing kind of fast so we can get to
+the 1.0 release, we got a bit sloppy with deprecation this release - sorry about that. We are very
+much trying to get to a place where we can commit to our APIs and stabilize the codebase, that is
+the primary goal of development at the moment. If you have API changing suggestions or requests
+please get them into us now so your needs can be met.
+
+Without further ado, here is the upgrade guide. Enjoy!
+
+## Suggested steps
+
+We suggest that you take these steps when upgrading:
+
+0. Make sure to update other dependency versions in `Cargo.toml` if you use them explicitly: `bitcoin_hashes` to 0.12.0, `secp256k1` to 0.27.0
+1. Remove all occurrences of `util::` referring to our crate
+2. Replace `Script` with `ScriptBuf` (`s/\([^A-Za-z0-9]\)Script\([^A-Za-z0-9]\)/\1ScriptBuf\2/g` should work in most cases)
+3. Replace instances of `.parse::<Address>()` with `.parse::<Address<_>>()`
+4. Call `require_network(network)` on parsed addresses (you'll get no method found for `Address<NetworkUnchecked>` errors)
+5. Replace `locktime` with `locktime::absolute`
+6. Replace `PackedLockTime` with just `LockTime`
+7. Import key types from the `key` submodule rather than `schnorr` or `ecdsa`
+8. Replace `SchnorrSighashType` with `TapSighashType`
+9. Replace `TapBranchHash` with `TapNodeHash`
+10. Change `hash_newtype!(FooHash, sha256::Hash, 32, doc="A hash of foo.");` to:
+
+```rust
+hash_newtype! {
+    /// A hash of foo.
+    pub struct FooHash(sha256::Hash);
+}
+```
+
+11. Fix outstanding compiler errors, if any
+12. Optimize the code: replace occurrences of `&ScriptBuf` with `Script`, remove allocations...
+13. Remove useless conversions `LockTime` -> `LockTime` (clippy has a lint for it)
+
+## Re-exports
+
+We are trying to separate the API from the directory structure, as part of this we are attempting to
+only re-export from the crate root types that exist in the standard Bitcoin vernacular i.e., for
+pubkey you can `use bitcoin::PublicKey` but to get an x-only pubkey you need to go to the `key`
+module `use bitcoin::key::XOnlyPublicKey`.
+
+Please note this is still work-in-progress, suggestions welcome.
+
+## Code moves
+
+We moved _a lot_ of stuff around. This was a precursor to crate smashing which we have now started.
+Hopefully things are in intuitive places, it might be useful to take a quick look at the new
+module structure to get a feel for things, specifically:
+
+- We have a workspace now! The main crate now lives in `bitcoin/`. The `bitcoin_hashes` repository
+  has been merged into the `rust-bitcoin` repository and now lives under `hashes/`.
+
+- The `util` module is all but gone, try just removing `util::` at first, most modules are
+  re-exported at the crate root.
+
+- Cryptography related stuff is now in the `crypto` module, submodules (excl. `taproot`) are
```

Comment:
Why does this still mention `crypto` module?

Reply:
Changed to this:

- Cryptography related stuff can now primarily be found in 4 modules (`ecdsa`, `taproot`, `sighash`,
  `key`). We have started to break ECDSA stuff up into legacy and segwit v0 where it is cleaner but
  this is still work-in-progress.